idea/src/org/jetbrains/kotlin/idea/intentions/ConvertLambdaToReferenceIntention.kt

```diff
@@ -220,13 +220,21 @@ open class ConvertLambdaToReferenceIntention(textGetter: () -> String) : SelfTar
 
     companion object {
         private fun KtLambdaExpression.lambdaParameterType(context: BindingContext? = null): KotlinType? {
-            val argument = parent as? KtValueArgument ?: return null
+            val argument = parentValueArgument() ?: return null
             val callExpression = argument.getStrictParentOfType<KtCallExpression>() ?: return null
             return callExpression
                 .getResolvedCall(context ?: analyze(BodyResolveMode.PARTIAL))
                 ?.getParameterForArgument(argument)?.type
         }
 
+        private fun KtLambdaExpression.parentValueArgument(): KtValueArgument? {
```

Comment:
There's an existing `unwrapIfLabeled()` utility function for this:
https://github.com/JetBrains/kotlin/blob/master/idea/src/org/jetbrains/kotlin/idea/intentions/loopToCallChain/commonUtils.kt#L151